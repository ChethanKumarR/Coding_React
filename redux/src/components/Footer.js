import React from "react";


const Footer=()=>{
    return(
        <header>
            <div>
               <hr/>
               &copy; Developer Chethan
            </div>
        </header>
    )
}
export default Footer ;