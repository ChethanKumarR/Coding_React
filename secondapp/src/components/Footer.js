import React from "react";


const Footer = () => {
    return (
        <div>
            <hr />
            <center>
                <h2>&copy;React 2020</h2>
            </center>
        </div>
    )
}
export default Footer;  