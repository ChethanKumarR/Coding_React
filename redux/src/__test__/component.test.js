import React from "react";
//importing create becoz we are doing functional or snapshot testing
import {create} from "react-test-renderer";
import BaseButton from "../components/BaseButton";
import Footer from "../components/Footer";

describe("Functional Testing", () => {

    //we can write multiple test cases like this
    ////functional testing
    test("button test", () => {
        const component = create(<BaseButton/>);
        const instance = component.root;
        const button = instance.findByType("button");
        button.props.onClick();
        expect(button.props.children).toBe("Unit Testing");
    });

    //snapshot testing
    test("snapshot Testing", () => {
        let tree = create(<Footer/>);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});