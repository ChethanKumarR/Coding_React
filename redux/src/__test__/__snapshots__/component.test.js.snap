// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functional Testing snapshot Testing 1`] = `
<header>
  <div>
    <hr />
    Â© Developer Chethan
  </div>
</header>
`;
